src/system-alloc.cc

```diff
@@ -545,11 +554,19 @@ bool TCMalloc_SystemRelease(void* start, size_t length) {
     return result != -1;
   }
 #endif
+#endif 
   return false;
 }
 
 void TCMalloc_SystemCommit(void* start, size_t length) {
+#if defined(FREE_MMAP_PROT_NONE) && defined(HAVE_MMAP)
+  // remaping as MAP_FIXED to same address assuming span size did not change 
+  // since last TCMalloc_SystemRelease
+  mmap(start, length, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS|MAP_FIXED,
+       -1, 0);
+#elif defined(MADV_FREE)
```

Comment:
one litttle nit-pick here, why bother with elif if you can simply do else ?

Reply:
Done